### PR TITLE
Fix `EventPayload` and `EventDelivery` creation

### PR DESCRIPTION
**Pull request by przlada**
I want to merge this change as it ensure atomic creation of `EventPayload` and related `EventDelivery`. This will prevent inconsistent DB state where payloads exist without deliveries what can lead to errors

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

File: saleor/core/tasks.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def delete_event_payloads_task(expiration_date=None):
     payloads_to_delete = EventPayload.objects.filter(
         ~Exists(valid_deliveries.filter(payload_id=OuterRef("id")))
     ).order_by("-pk")
-    ids = payloads_to_delete.values_list("pk", flat=True)[:BATCH_SIZE]
+    ids = list(payloads_to_delete.values_list("pk", flat=True)[:BATCH_SIZE])
```
Comment (kadewu) [inline saleor/core/tasks.py:37]: Is that change required as well?

Comment (przlada) [reply saleor/core/tasks.py:37]: I forgot to mention, this is a related, but separate bug.
Without it, the QuerySet is reevaluated on every access and can lead to a different set of payloads being fetched for file names and a different set for deletion

**Review by kadewu — APPROVED** [review]
